core/src/main/scala/org/apache/spark/crypto/CommonConfigurationKeys.scala

```diff
@@ -0,0 +1,57 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.crypto
+
+import org.apache.hadoop.io.Text
+
+import com.intel.chimera.codec.CipherSuite
+import com.intel.chimera.codec.JceAesCtrCryptoCodec
+
+/**
+ * Constants and variables for spark shuffle encryption
+ */
+private[spark] object CommonConfigurationKeys {
```

Comment:
Is any of this necessary, since all config options are defined by the library?

You could just do a wholesale translation of config options from SparkConf to Chimera's configuration, something like:

    def toChimeraConf(conf: SparkConf, prefix: String): Properties = {
      val props = new Properties()
      conf.getAll.foreach { case (k, v) =>
        if (v.startsWith(prefix)) {
          props.putProperty(k.substring(prefix.length()), v)
        }
      }
      props
    }

That way `spark.shuffle.chimera.foo` becomes `chimera.foo`; no need to duplicate anything that is defined in the Chimera library (except for documentation under `docs/` so users know how to configure this).